src/UsgsAstroFrameSensorModel.cpp

```diff
@@ -900,68 +922,95 @@ std::string UsgsAstroFrameSensorModel::constructStateFromIsd(const std::string&
+    std::vector<double> sunPositions = {};
+
+    for (int i = 0; i < ephemTime.size(); i++) {
+      rotatedSunState = j2000_to_target.rotateStateAt(ephemTime[i], sunStates[i]);
+      sunPositions.push_back(rotatedSunState.position.x * 1000);
+      sunPositions.push_back(rotatedSunState.position.y * 1000);
+      sunPositions.push_back(rotatedSunState.position.z * 1000);
+    }
+
+    state["m_sunPosition"] = sunPositions;
 
     // get sensor_orientation quaternion
-    std::vector<double> quaternion = getSensorOrientations(isd, parsingWarnings);
-    if (quaternion.size() != 4) {
+    ale::Orientations j2000_to_sensor = ale::getInstrumentPointing(parsedIsd);
+    ale::Orientations sensor_to_j2000 = j2000_to_sensor.inverse();
+    ale::Orientations sensor_to_target = j2000_to_target * sensor_to_j2000;
+    ephemTime = sensor_to_target.getTimes();
+    std::vector<double> quaternion;
+    std::vector<double> quaternions;
+
+    for (int i = 0; i < ephemTime.size(); i++) {
+      ale::Rotation rotation = sensor_to_target.interpolate(ephemTime[i], ale::SLERP);
+      quaternion = rotation.toQuaternion();
+      quaternions.push_back(quaternion[1]);
+      quaternions.push_back(quaternion[2]);
+      quaternions.push_back(quaternion[3]);
+      quaternions.push_back(quaternion[0]);
+    }
+
+    if (quaternions.size() != 4) {
       parsingWarnings->push_back(
         csm::Warning(
           csm::Warning::DATA_NOT_AVAILABLE,
           "Sensor orientation quaternion does not have 4 values.",
           "UsgsAstroFrameSensorModel::constructStateFromIsd()"));
     }
     else {
-      state["m_currentParameterValue"][3] = quaternion[0];
-      state["m_currentParameterValue"][4] = quaternion[1];
-      state["m_currentParameterValue"][5] = quaternion[2];
-      state["m_currentParameterValue"][6] = quaternion[3];
+      state["m_currentParameterValue"][3] = quaternions[0];
+      state["m_currentParameterValue"][4] = quaternions[1];
+      state["m_currentParameterValue"][5] = quaternions[2];
+      state["m_currentParameterValue"][6] = quaternions[3];
     }
 
     // get optical_distortion
-    state["m_distortionType"] = getDistortionModel(isd, warnings);
-    state["m_opticalDistCoeffs"] = getDistortionCoeffs(isd, warnings);
+    state["m_distortionType"] = getDistortionModel(ale::getDistortionModel(parsedIsd));
+    state["m_opticalDistCoeffs"] = ale::getDistortionCoeffs(parsedIsd);
 
     // get detector_center
-    state["m_ccdCenter"][0] = getDetectorCenterLine(isd, parsingWarnings);
-    state["m_ccdCenter"][1] = getDetectorCenterSample(isd, parsingWarnings);
+    state["m_ccdCenter"][0] = ale::getDetectorCenterLine(parsedIsd);
+    state["m_ccdCenter"][1] = ale::getDetectorCenterSample(parsedIsd);
 
 
     // get radii
-    state["m_minorAxis"] = getSemiMinorRadius(isd, parsingWarnings);
-    state["m_majorAxis"] = getSemiMajorRadius(isd, parsingWarnings);
+    state["m_minorAxis"] = ale::getSemiMinorRadius(parsedIsd) * 1000;
```

Comment:
add a quick comment about units that require the * 1000 (ex: in km, but state needs to be in meters)?